### PR TITLE
Nested layout

### PR DESCRIPTION
**Pull request by T-Wizard**
Add Support for Components in Components (#181)

A simpler more limited alternative to https://github.com/qlik-oss/picasso.js/compare/master...nilzona:compose-api

**Checklist**

- [ ] tests added
- [ ] commits conform to the [commit guidelines](./CONTRIBUTING.md#commit)
- [ ] documentation updated

### CODE REVIEW

**Comment by nilzona** [comment]
@T-Wizard  This looks promising. Do you think you can add an example picasso configuration to try it out?

**Comment by T-Wizard** [comment]
@nilzona: added two examples in the studio

File: packages/picasso.js/src/core/chart/component-collection.js [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+    const dockConfig = c.instance.dockConfig();
+    return {
+      index: i,
+      key: c.settings.key,
+      dockConfig,
+      resize: c.instance.resize,
+      preferredSize(opts) {
+        return dockConfig.computePreferredSize({ ...opts, children: wrapChildren(c.children) });
+      },
+    };
```
Comment (T-Wizard) [inline packages/picasso.js/src/core/chart/component-collection.js:64]: any suggestion on what information is needed by different types of layout functions?

File: packages/picasso.js/src/core/chart/index.js [hunk 2]
```diff
@@ -35,6 +35,8 @@ import themeFn from '../theme';
  * @property {boolean} [show = true] If the component should be rendered
  * @property {string} [scale] Named scale. Will be provided to the component if it ask for it.
  * @property {string} [formatter] Named formatter. Fallback to create formatter from scale. Will be provided to the component if it ask for it.
+ * @property {ComponentSettings[]} [components] Optional list of child components
+ * @property {DockLayoutSettings|function} [strategy] Layout strategy used for child components.
```
Comment (T-Wizard) [inline packages/picasso.js/src/core/chart/index.js:39]: it is possible to make properties as experimental when defined like this?

File: packages/picasso.js/src/core/layout/dock/docker.js [hunk 3]
```diff
@@ -415,7 +406,7 @@ function dockLayout(initialSettings) {
       throw new Error('Invalid rect');
     }
     if (!components.length) {
-      return { visible: [], hidden: [] };
+      return { visible: [], hidden: [], order: [] };
```
Comment (veinfors) [inline packages/picasso.js/src/core/layout/dock/docker.js:409]: `order` should be `ordered` here, right?

Comment (T-Wizard) [reply packages/picasso.js/src/core/layout/dock/docker.js:409]: yes

**Review by veinfors — APPROVED** [review]

File: packages/picasso.js/src/core/chart-components/container/container.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
+/**
+ * @typedef {object} ContainerComponent
```
Comment (cbt1) [inline packages/picasso.js/src/core/chart-components/container/container.js:2]: Rename to `ComponentContainer` to be consistent with other components

Comment (T-Wizard) [reply packages/picasso.js/src/core/chart-components/container/container.js]: done

File: docs/scriptappy.json [hunk 5]
```diff
@@ -1973,6 +2017,25 @@
         }
       }
     },
+    "ContainerComponent": {
```
Comment (cbt1) [inline docs/scriptappy.json:2020]: You forgot to build the spec

Comment (T-Wizard) [reply docs/scriptappy.json]: done
(many changes in the spec, probably not any rebuild since the last rebase from master)

**Review by cbt1 — APPROVED** [review]

**Pull request merged**